### PR TITLE
ENH as_frame defaults to True for all fetch_*

### PR DESCRIPTION
**Pull request by shimst3r**
## Description

As suggested by @romanlutz in #1206, this PR sets the default value for `as_frame=True` for all fetch functions.

## Tests

- [x] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [x] user guide added or updated
- [x] API docs added or updated
- [x] example notebook added or updated

### CODE REVIEW

**Comment by shimst3r** [comment]
Sorry for the swathe of whitespace changes in `docs/user_guide/datasets/boston_housing_data.rst`, my formatter seemed to be a bit too eager in that case. 😅

File: fairlearn/datasets/_fetch_acs_income.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def fetch_acs_income(
         By default, all fairlearn data is stored in '~/.fairlearn-data'
         subfolders.
 
-    as_frame : bool, default=False
+    as_frame : bool, default=True
```
Comment (romanlutz) [inline fairlearn/datasets/_fetch_acs_income.py:50]: Can you add the versionchanged tag with a brief note on the default changing? Same in the other files.

**Review by romanlutz — APPROVED** [review]
Amazing! I should file more issues, they magically get done @shimst3r 🤣 jokes aside, thank you for adjusting all the references.

**Comment by shimst3r** [comment]
> I should file more issues, they magically get done @shimst3r 🤣

Better don't try this at work though, might have unexpected side-effects. 😂

File: fairlearn/datasets/_fetch_acs_income.py (hunk 1 above)
Comment (shimst3r) [reply fairlearn/datasets/_fetch_acs_income.py:50]: Yep! Took the "no deprecation cycle" a bit too literally I guess. 😄

File: fairlearn/datasets/_fetch_acs_income.py [hunk 2]
```diff
@@ -55,6 +55,11 @@ def fetch_acs_income(
         data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
         DataFrames or Series as describe above.
 
+        .. note::
+            Default value changed to True.
+
+        .. versionchanged:: 0.9.0
```
Comment (romanlutz) [inline fairlearn/datasets/_fetch_acs_income.py:61]: nit: you can actually combine the note and versionchanged!
See https://github.com/fairlearn/fairlearn/blob/c09c2ea4a1f6c4c25d0941212b59cf7475f4e518/fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py for example. The note just goes right under versionchanged. I believe they end up in the same box that way. OK, done nitpicking now 😆

Comment (shimst3r) [reply fairlearn/datasets/_fetch_acs_income.py]: Oooh that's actually really nice. 🤓 Never stop nitpicking as long as there's room for improvement, haha.

**Pull request merged**